allennlp/common/plugins.py

```diff
@@ -0,0 +1,74 @@
+"""
+Plugin management.
+
+AllenNLP supports loading "plugins" dynamically. A plugin is just a Python package that is found
+by AllenNLP with the methods provided in this module.
+
+There are two ways of declaring plugins for discovery:
+
+    * Writing the package name to import in a file (typically ".allennlp_plugins", in the current
+    directory from which the command "allennlp" is run). This is the simplest approach.
+
+    * Creating a folder called "allennlp_plugins" that's in the Python path when you run the
+    "allennlp" command (typically under your project's root directory), then creating a subfolder
+    with the name you want and creating an "__init.py" file that imports the code you want (e.g.,
+    your Python package). This option is preferred when you want to create a pip-installable
+    package and you want to make your AllenNLP plugin available when users install your package.
```

Comment:
Can you give an example of this?

Reply:
Oh, what I meant was an example of *how* to do this.  If that example is a pointer to a repo that does it, that's fine too. (and in this case it's ok to put a pointer to a repo that doesn't do it yet, but will very soon)

Reply:
Haha, I wonder why don't you just say the repo name.